save-orchestrator/src/main/kotlin/com/saveourtool/save/orchestrator/config/KafkaConfiguration.kt

```diff
@@ -0,0 +1,164 @@
+package com.saveourtool.save.orchestrator.config
+
+import com.saveourtool.save.kafka.TestExecutionTaskDto
+import com.saveourtool.save.orchestrator.kafka.AgentKafkaListener
+import com.saveourtool.save.orchestrator.kafka.KafkaSender
+import org.apache.kafka.clients.admin.NewTopic
+import org.apache.kafka.clients.consumer.ConsumerConfig
+import org.apache.kafka.clients.consumer.CooperativeStickyAssignor
+import org.apache.kafka.clients.producer.ProducerConfig
+import org.apache.kafka.clients.producer.internals.DefaultPartitioner
+import org.apache.kafka.common.serialization.StringDeserializer
+import org.apache.kafka.common.serialization.StringSerializer
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.boot.autoconfigure.kafka.KafkaProperties
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+import org.springframework.context.annotation.Profile
+import org.springframework.kafka.config.ConcurrentKafkaListenerContainerFactory
+import org.springframework.kafka.core.*
+import org.springframework.kafka.listener.DeadLetterPublishingRecoverer
+import org.springframework.kafka.listener.SeekToCurrentErrorHandler
+import org.springframework.kafka.support.serializer.ErrorHandlingDeserializer
+import org.springframework.kafka.support.serializer.JsonDeserializer
+import org.springframework.kafka.support.serializer.JsonSerializer
+import java.util.*
+
+/**
+ * Kafka producer and consumer configuration
+ *
+ * @property kafkaProperties
+ */
+@Configuration
+@Profile("dev & kafka")
```

Comment:
what is profile "kafka" for?

Reply:
kafka is for kafka specific settings like bootstrap servers, topics etc, it can be quite a lot so it's better to keep it in separate file